src/csi-wrapper/pkg/wrapper/controllerservice.go

```diff
@@ -362,6 +367,13 @@ func (s *ControllerService) SyncHandler(peerPodVolume *peerpodvolumeV1alpha1.Pee
 	if peerPodVolume.Status.State == peerpodvolumeV1alpha1.PeerPodVSIIDReady && peerPodVolume.Spec.DevicePath == "" {
 		// After peerpod vsi id is ready in crd object, we can reproduce the ControllerPublishVolumeRequest
 		vsiID := peerPodVolume.Spec.VMID
+
+		// The azure csi driver requires the nodeID to be just the name of the VM,
+		// not the full Azure Resource ID, as it is saved in the PeerpodVolume object
+		if azureVMRegexp.MatchString(vsiID) {
```

Comment:
Do we expect situations where VMID wouldn't be full resource id?

Reply:
Not for Azure, but other providers might have VM IDs containing slashes, and also have a CSI driver which expects that full VM ID with slashes

Reply:
I see. any chance we don't have to rely on heuristics to be aware of the cloudprovider here?

Reply:
I don't think so.
As far as I can tell the, neither the peerpod volume spec nor the ControllerService save the active cloud provider anywhere.